### PR TITLE
Add typings for TypedArrayUtils.Kdtree

### PR DESCRIPTION
**Pull request by ethankay**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [NA] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: // https://github.com/mrdoob/three.js/blob/master/examples/js/TypedArrayUtils.js
- [NA] Increase the version number in the header if appropriate.
- [NA] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/three/three-typedarrayutils.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      left: Node;
+      right: Node;
+      parent: Node;
+      depth: number;
+      pos: number;
+    }
+  }
+
+  export class Kdtree {
+    constructor(points: Float32Array, metric: any, eleSize: number);
```
Comment (PranavSenthilnathan) [inline types/three/three-typedarrayutils.d.ts]: Does it make sense to type `metric` as `(a: Float32Array, b: Float32Array) => number`?

File: types/three/three-typedarrayutils.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// https://github.com/mrdoob/three.js/blob/master/examples/js/TypedArrayUtils.js
+
+export namespace TypedArrayUtils {
```
Comment (PranavSenthilnathan) [inline types/three/three-typedarrayutils.d.ts]: Tests would be great to show how using this in TS would look.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/three/three-typedarrayutils.d.ts (hunk 2 above)
Comment (ethankay) [reply types/three/three-typedarrayutils.d.ts]: The doc comments in the original have a decent minimal example.

File: types/three/three-typedarrayutils.d.ts (hunk 1 above)
Comment (PranavSenthilnathan) [reply types/three/three-typedarrayutils.d.ts]: I'm just going by the call in the JS code which is `metric( point, node.obj )` and the type of `node.obj`from this d.ts is `obj: Float32Array;`.

**Comment by ethankay** [comment]
@PranavSenthilnathan Can you please review?

**Comment by orta** [comment]
Three doesn't exist in DefinitelyTyped anymore, I think they must be hosting their own types. Sorry this PR got a bit lost but I'm going to close it.

**Pull request closed without merging**